### PR TITLE
Add more options to style change

### PR DESCRIPTION
**Pull request by pacozaa**
Add more options to style change when `CNToolbar` use only a few options.

### CODE REVIEW

**Comment by pacozaa** [comment]
Add textInputStyle props to document.

**Comment by mqtik** [comment]
I'm trying to use this, but the changes are looking for a iconSet prop to perform a map.

But the prop is undefined.

How would you set this?

**Comment by pacozaa** [comment]
@mqtik  checkout my new commit  :)

File: src/CNToolbar.js [hunk 1]
```diff
@@ -1,385 +1,111 @@
...
+                                selectedColor={selectedColor ? selectedColor : defaultSelectedColor}
+                                selectedBackgroundColor={selectedBackgroundColor ? selectedBackgroundColor : defaultSelectedBgColor}
+                                iconArray={object.iconArray}
+                                iconSetContainerStyles={[styles.iconSetContainer, iconSetContainer]}
+                                iconStyles={[styles.iconContainer, iconContainerStyle]}
+                                onStyleKeyPress={this.onStyleKeyPress}
+                            /> :
+                            <CNSeperator
+                                key={index}
+                            />
```
Comment (imnapo) [inline src/CNToolbar.js]:
```suggestion
                            color={color || defaultColor}
                            />
```

File: src/CNToolbarIcon.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        iconComponent,
+        onStyleKeyPress,
+        selectedColor,
+        selectedStyles,
+        selectedTag,
+        buttonTypes,
+        selectedBackgroundColor,
+    } = props
+    let backgroundColorCondition = ''
+
```
Comment (imnapo) [inline src/CNToolbarIcon.js]:
```suggestion
let colorCondition = ''
```

File: src/CNToolbarIcon.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    if (buttonTypes === 'style') {
+        backgroundColorCondition = selectedStyles.indexOf(toolTypeText) >= 0 ? selectedBackgroundColor : backgroundColor
+    }
```
Comment (imnapo) [inline src/CNToolbarIcon.js]:
```suggestion
    colorCondition = selectedStyles.indexOf(toolTypeText) >= 0 ? selectedColor : color
    }
```

File: src/CNToolbarIcon.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    else if (buttonTypes === 'tag') {
+        backgroundColorCondition = selectedTag === toolTypeText ? selectedBackgroundColor : backgroundColor
+    }
```
Comment (imnapo) [inline src/CNToolbarIcon.js]:
```suggestion
    colorCondition = selectedTag === toolTypeText ? selectedColor : color
    }
```

File: src/CNToolbarIcon.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+                onStyleKeyPress(toolTypeText)
+            }}
+        >
+            <View style={[iconStyles,
+                {
+                    backgroundColor: backgroundColorCondition
+                }]}
+            >
+                {
+                    React.cloneElement(iconComponent, { size, color: selectedStyles.indexOf(toolTypeText) >= 0 ? selectedColor : color })
```
Comment (imnapo) [inline src/CNToolbarIcon.js]:
```suggestion
                   React.cloneElement(iconComponent, { size, color: colorCondition})
```

**Comment by mqtik** [comment]
Excellent, I will try it.

I'm preparing a few pull request too and I can't merge just yet.

File: src/CNSeperator.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import React, { Component } from 'react';
+import {
+    View,
+    TouchableWithoutFeedback,
+    TouchableHighlight,
+    Text,
+    StyleSheet
+} from 'react-native'
+
+const defaultSelectedBgColor = '#e4e4e4'
```
[the remark this answers is not in the document]
Comment (pacozaa) [reply src/CNSeperator.js]: Looks good. I originally use this because of personal project :)

**Comment by pacozaa** [comment]
@imnapo I resolved files as suggested but still haven't run an example yet. Do you have a set of command to run `expo-demo`? I will help add it to repo too.

**Pull request merged**